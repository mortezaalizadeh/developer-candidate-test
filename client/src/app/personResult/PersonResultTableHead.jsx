import React, { Component } from 'react';
import PropTypes from 'prop-types';
import TableCell from '@material-ui/core/TableCell';
import TableHead from '@material-ui/core/TableHead';
import TableRow from '@material-ui/core/TableRow';
import TableSortLabel from '@material-ui/core/TableSortLabel';
import Tooltip from '@material-ui/core/Tooltip';
import { withTranslation } from 'react-i18next';

class PersonResultTableHead extends Component {
  constructor(props, context) {
    super(props, context);

    const { t } = this.props;

    this.state = {
      columns: [
        { id: 'name', numeric: false, disablePadding: true, label: t('columns.name') },
        { id: 'gender', numeric: true, disablePadding: false, label: t('columns.gender') },
        { id: 'age', numeric: true, disablePadding: false, label: t('columns.age') },
      ],
    };
  }
  createSortHandler = property => event => {
    const { onRequestSort } = this.props;

    onRequestSort(event, property);
  };

  render = () => {
    const { sortOrder, sortColumn } = this.props;
    const { columns } = this.state;

    return (
      <TableHead>
        <TableRow>
          <TableCell padding="checkbox" />
          {columns.map(
            column => (
              <TableCell
                key={column.id}
                align={column.numeric ? 'right' : 'left'}
                padding={column.disablePadding ? 'none' : 'default'}
                sortDirection={sortColumn === column.id ? sortOrder : false}>
                <Tooltip title="Sort" placement={column.numeric ? 'bottom-end' : 'bottom-start'} enterDelay={300}>
                  <TableSortLabel active={sortColumn === column.id} direction={sortOrder} onClick={this.createSortHandler(column.id)}>
                    {column.label}
                  </TableSortLabel>
                </Tooltip>
              </TableCell>
            ),
            this,
          )}
        </TableRow>
      </TableHead>
    );
  };
}

PersonResultTableHead.propTypes = {
  onRequestSort: PropTypes.func.isRequired,
  sortOrder: PropTypes.string.isRequired,
  sortColumn: PropTypes.string.isRequired,
  t: PropTypes.func.isRequired,
};

export default withTranslation()(PersonResultTableHead);
