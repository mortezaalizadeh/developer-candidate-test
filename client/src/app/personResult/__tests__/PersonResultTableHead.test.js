import React from 'react';
import renderer from 'react-test-renderer';
import PersonResultTableHead from '../PersonResultTableHead';

it('should render correctly', () => {
  const tree = renderer.create(<PersonResultTableHead sortColumn="name" sortOrder="asc" />).toJSON();

  expect(tree).toMatchSnapshot();
});
