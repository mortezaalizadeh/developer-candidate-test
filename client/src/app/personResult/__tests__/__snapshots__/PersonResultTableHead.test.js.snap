// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<thead
  className="MuiTableHead-root-1"
>
  <tr
    className="MuiTableRow-root-2 MuiTableRow-head-5"
  >
    <th
      aria-sort={null}
      className="MuiTableCell-root-7 MuiTableCell-head-8 MuiTableCell-paddingCheckbox-13"
      scope="col"
    />
    <th
      aria-sort="ascending"
      className="MuiTableCell-root-7 MuiTableCell-head-8 MuiTableCell-alignLeft-15 MuiTableCell-paddingNone-14"
      scope="col"
    >
      <span
        aria-describedby={null}
        className="MuiButtonBase-root-32 MuiTableSortLabel-root-27 MuiTableSortLabel-active-28"
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseOver={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex="0"
        title="Sort"
      >
        columns.name
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root-35 MuiTableSortLabel-icon-29 MuiTableSortLabel-iconDirectionAsc-31"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
          />
        </svg>
      </span>
    </th>
    <th
      aria-sort={null}
      className="MuiTableCell-root-7 MuiTableCell-head-8 MuiTableCell-alignRight-17"
      scope="col"
    >
      <span
        aria-describedby={null}
        className="MuiButtonBase-root-32 MuiTableSortLabel-root-27"
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseOver={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex="0"
        title="Sort"
      >
        columns.gender
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root-35 MuiTableSortLabel-icon-29 MuiTableSortLabel-iconDirectionAsc-31"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
          />
        </svg>
      </span>
    </th>
    <th
      aria-sort={null}
      className="MuiTableCell-root-7 MuiTableCell-head-8 MuiTableCell-alignRight-17"
      scope="col"
    >
      <span
        aria-describedby={null}
        className="MuiButtonBase-root-32 MuiTableSortLabel-root-27"
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseOver={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex="0"
        title="Sort"
      >
        columns.age
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root-35 MuiTableSortLabel-icon-29 MuiTableSortLabel-iconDirectionAsc-31"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
          />
        </svg>
      </span>
    </th>
  </tr>
</thead>
`;
